engine/lib/worker/exit_controller.go

```diff
@@ -71,6 +127,14 @@ func (c *ExitController) PollExit() error {
 		return nil
 	}
 
+	c.callPrepareOnce.Do(func() {
```

Comment:
Why we need another sync.Once here? w.closeImplOnce can't achieve the `once` precisely?

Reply:
The one in `ExitController` is for redundancy. I can change it to using an atomic, because there is no risk of racing to calling the `prepareExitFunc`.